spot-client/src/common/remote-control/P2PSignalingBase.js

```diff
@@ -325,19 +325,23 @@ export default class P2PSignalingBase extends Emitter {
      * @returns {void}
      */
     _processOffer(from, data) {
-        if (this.getConnectionForAddress(from)) {
-            logger.error('Received offer, but PeerConnection is still there', {
-                data,
-                from
-            });
+        const offer = this._parseJSONorLogError(data.offer, 'offer message');
 
+        if (!offer) {
             return;
         }
 
-        const offer = this._parseJSONorLogError(data.offer, 'offer message');
+        const connection = this.getConnectionForAddress(from);
+
+        if (connection && connection.isDataChannelActive()) {
+            logger.error('Received offer, but PeerConnection is still there', {
+                data,
+                from
+            });
 
-        if (!offer) {
             return;
+        } else if (connection) {
+            this.closeConnection(from);
```

Comment:
In this case the other side of the connection should already have closed its p2p? So this is more of a cleanup step for a dangling connection? Is there a chance that closing this connection could somehow signal the other side to close an active connection?

Reply:
Yes it is a cleanup for dangling connection. It should have already closed it otherwise would not be able to send a new offer. Anyway on that other side it would get "disconnected" ICE state when it's closed here, but it should not really happen.